MeshForwarder::LogIp6Message not building

Current master is not building the LogIp6Message function when OPENTHREAD_CONFIG_LOG_LEVEL >= OPENTHREAD_LOG_LEVEL_INFO, which in my case is 4 > 3. Probably introduced in  #1515 ?

Compiler spits out:
>   CXX      thread/libopenthread_mtd_a-mesh_forwarder.o
> /home/gr3yh0und/Downloads/openthread/examples/../src/core/thread/mesh_forwarder.cpp: In member function 'void Thread::MeshForwarder::LogIp6Message(Thread::MeshForwarder::MessageAction, const Thread::Message&, const Thread::Mac::Address&, ThreadError)':
> /home/gr3yh0und/Downloads/openthread/examples/../src/core/thread/mesh_forwarder.cpp:2286:14: error: variable 'checksum' set but not used [-Werror=unused-but-set-variable]
>      uint16_t checksum = 0;
>               ^
> /home/gr3yh0und/Downloads/openthread/examples/../src/core/thread/mesh_forwarder.cpp:2289:10: error: unused variable 'stringBuffer' [-Werror=unused-variable]
>      char stringBuffer[Ip6::Address::kIp6AddressStringSize];
>           ^
> /home/gr3yh0und/Downloads/openthread/examples/../src/core/thread/mesh_forwarder.cpp: At global scope:
> /home/gr3yh0und/Downloads/openthread/examples/../src/core/thread/mesh_forwarder.cpp:2283:49: error: unused parameter 'aAction' [-Werror=unused-parameter]
>  void MeshForwarder::LogIp6Message(MessageAction aAction, const Message &aMessage, const Mac::Address &aMacAddress,
>                                                  ^
> /home/gr3yh0und/Downloads/openthread/examples/../src/core/thread/mesh_forwarder.cpp:2283:103: error: unused parameter 'aMacAddress' [-Werror=unused-parameter]
>  void MeshForwarder::LogIp6Message(MessageAction aAction, const Message &aMessage, const Mac::Address &aMacAddress,
>                                                                                                        ^
> /home/gr3yh0und/Downloads/openthread/examples/../src/core/thread/mesh_forwarder.cpp:2284:47: error: unused parameter 'aError' [-Werror=unused-parameter]
>                                    ThreadError aError)
>                                                ^
> /home/gr3yh0und/Downloads/openthread/examples/../src/core/thread/mesh_forwarder.cpp: In member function 'void Thread::MeshForwarder::LogIp6Message(Thread::MeshForwarder::MessageAction, const Thread::Message&, const Thread::Mac::Address&, ThreadError)':
> /home/gr3yh0und/Downloads/openthread/examples/../src/core/thread/mesh_forwarder.cpp:2286:14: error: variable 'checksum' set but not used [-Werror=unused-but-set-variable]
>      uint16_t checksum = 0;
>               ^
> /home/gr3yh0und/Downloads/openthread/examples/../src/core/thread/mesh_forwarder.cpp:2289:10: error: unused variable 'stringBuffer' [-Werror=unused-variable]
>      char stringBuffer[Ip6::Address::kIp6AddressStringSize];
>           ^
> /home/gr3yh0und/Downloads/openthread/examples/../src/core/thread/mesh_forwarder.cpp: At global scope:
> /home/gr3yh0und/Downloads/openthread/examples/../src/core/thread/mesh_forwarder.cpp:2283:49: error: unused parameter 'aAction' [-Werror=unused-parameter]
>  void MeshForwarder::LogIp6Message(MessageAction aAction, const Message &aMessage, const Mac::Address &aMacAddress,
>                                                  ^
> /home/gr3yh0und/Downloads/openthread/examples/../src/core/thread/mesh_forwarder.cpp:2283:103: error: unused parameter 'aMacAddress' [-Werror=unused-parameter]
>  void MeshForwarder::LogIp6Message(MessageAction aAction, const Message &aMessage, const Mac::Address &aMacAddress,
>                                                                                                        ^
> /home/gr3yh0und/Downloads/openthread/examples/../src/core/thread/mesh_forwarder.cpp:2284:47: error: unused parameter 'aError' [-Werror=unused-parameter]
>                                    ThreadError aError)
>                                                ^
> cc1plus: all warnings being treated as errors
> Makefile:3223: recipe for target 'thread/libopenthread_mtd_a-mesh_forwarder.o' failed
> make[4]: *** [thread/libopenthread_mtd_a-mesh_forwarder.o] Error 1

--- a/src/core/thread/mesh_forwarder.cpp
+++ b/src/core/thread/mesh_forwarder.cpp
@@ -2278,7 +2278,7 @@ void MeshForwarder::HandleDataPollTimeout(void)
     }
 }
 
-#if OPENTHREAD_CONFIG_LOG_LEVEL >= OPENTHREAD_LOG_LEVEL_INFO
+#if (OPENTHREAD_CONFIG_LOG_LEVEL >= OPENTHREAD_LOG_LEVEL_INFO) && (OPENTHREAD_CONFIG_LOG_MAC == 1)
 
 void MeshForwarder::LogIp6Message(MessageAction aAction, const Message &aMessage, const Mac::Address &aMacAddress,
                                   ThreadError aError)

